addons/point_of_sale/tests/test_report_session.py

```diff
@@ -112,3 +112,88 @@ def test_report_session_2(self):
+            'journal_id': self.company_data['default_journal_cash'].id,
+            'company_id': self.env.company.id,
+        })
+        bank_payment_method = self.env['pos.payment.method'].create({
+            'name': 'Bank',
+            'journal_id': self.company_data['default_journal_bank'].id,
+            'receivable_account_id': self.company_data['default_account_receivable'].id,
+            'company_id': self.env.company.id,
+        })
+        self.config.write({'payment_method_ids': [(4, bank_payment_method.id), (4, cash_payment_method.id)]})
+
+        self.open_new_session()
+        session = self.pos_session
+
+        order = self.env['pos.order'].create({
+            'session_id': session.id,
+            'lines': [(0, 0, {
+                'name': "TR/0001",
+                'product_id': product1.id,
+                'price_unit': 150,
+                'discount': 0,
+                'qty': 1.0,
+                'price_subtotal': 150,
+                'price_subtotal_incl': 165,
+            }), (0, 0, {
+                'name': "TR/0001",
+                'product_id': product2.id,
+                'price_unit': 150,
+                'discount': 0,
+                'qty': 1.0,
+                'price_subtotal': 150,
+                'price_subtotal_incl': 165,
+            })],
+            'amount_total': 330.0,
+            'amount_tax': 30.0,
+            'amount_paid': 0.0,
+            'amount_return': 0.0,
+        })
+        payment_context = {"active_ids": order.ids, "active_id": order.id}
+
+        order_payment = self.env['pos.make.payment'].with_context(**payment_context).create({
+            'amount': 65,
+            'payment_method_id': cash_payment_method.id
+        })
+        order_payment.with_context(**payment_context).check()
+        order_payment = self.env['pos.make.payment'].with_context(**payment_context).create({
+            'amount': 100,
+            'payment_method_id': cash_payment_method.id
+        })
+        order_payment.with_context(**payment_context).check()
+
+        order_payment = self.env['pos.make.payment'].with_context(**payment_context).create({
+            'amount': 100,
+            'payment_method_id': bank_payment_method.id
+        })
+        order_payment.with_context(**payment_context).check()
+        order_payment = self.env['pos.make.payment'].with_context(**payment_context).create({
+            'amount': 65,
+            'payment_method_id': bank_payment_method.id
+        })
+        order_payment.with_context(**payment_context).check()
+
+        order_report_lines = self.env['report.pos.order'].sudo().search([('order_id', '=', order.id)])
+
+        self.assertEqual(len(order_report_lines), 2)
+        self.assertEqual(order_report_lines[0].payment_method_id.id, order_report_lines[1].payment_method_id.id)
+
+        for order in order_report_lines:
+            self.assertEqual(order.price_total, 150.0)
```

Comment:
This goes with the other nitpicking comment, you can change it to 165. 

I'm making the suggestion as I see that `price_total` is defined using the order line `price_subtotal_incl` and you had it set to 165 in the order but since you don't put the tax information it gets actually set to 150.